tests/func_test.py

```diff
@@ -0,0 +1,65 @@
+from xdsl.ir import Block, Region
+from xdsl.dialects.func import FuncOp, Return
+from xdsl.dialects.arith import Addi, Constant
+from xdsl.dialects.builtin import IntegerAttr, i32
+
+
+def test_func():
+    # Create two constants and add them, add them in a region and
+    # create a function
+    a = Constant.from_attr(IntegerAttr.from_int_and_width(1, 32), i32)
+    b = Constant.from_attr(IntegerAttr.from_int_and_width(2, 32), i32)
+    # Operation to add these constants
+    c = Addi.get(a, b)
+
+    # Create a region to include a, b, c
+    region = Region.from_operation_list([a, b, c])
+
+    # Use this region to create a func0
+    func0 = FuncOp.from_region("func0", [], [], region)
```

Comment:
```suggestion
    func0 = FuncOp.from_region(
        "func0",
        [],
        [],
        Region.from_operation_list([
            a := Constant.from_int_constant(1, 32),  #
            b := Constant.from_int_constant(2, 32),
            Addi.get(a, b)
        ]))
```

IMO, this way of building regions is nicer as it looks closer to how the print would look like.

Reply:
I agree, but this is not the purpose here. Try to think of how a visitor would look starting from inside out, assuming we do not know the structure...

Reply:
Yes, sure. But I feel like my suggestion is easier to read for someone reading the test. In the end, we are testing the func dialect here, in particular the `from_region` constructor, so IMO there should be as little time spent thinking about how the region looks like as possible.

Reply:
I see; I think it is good to add both and also compare to ensure cross-validation!

Reply:
But this is something we should do as part of the test for the `Region.from_operation_list` constructor, right? So not as part of the func dialect tests.

Reply:
Yes, I agree with that. We want to build functionality that depends on each other, so we should properly unit tests parts of it. But yes, let's do that in a later PR.